src/mscorlib/corefx/System/Globalization/TextInfo.Unix.cs

```diff
@@ -107,13 +109,21 @@ internal unsafe void ChangeCase(char* src, int srcLen, char* dstBuffer, int dstB
             {
                 Interop.GlobalizationInterop.ChangeCaseInvariant(src, srcLen, dstBuffer, dstBufferCapacity, bToUpper);
             }
-            else if (m_needsTurkishCasing)
-            {
-                Interop.GlobalizationInterop.ChangeCaseTurkish(src, srcLen, dstBuffer, dstBufferCapacity, bToUpper);
-            }
             else
             {
-                Interop.GlobalizationInterop.ChangeCase(src, srcLen, dstBuffer, dstBufferCapacity, bToUpper);
+                if (!m_needsTurkishCasingSet)
+                {
+                  m_needsTurkishCasingSet = true;
+                  m_needsTurkishCasing = NeedsTurkishCasing(m_textInfoName);
```

Comment:
@stephentoub 
OK `m_needsTurkishCasingSet = true;` should be perfromed after `m_needsTurkishCasing = NeedsTurkishCasing(m_textInfoName);`

Other than that I suspect that CompareInfo.Compare should be trhead-safe when it comes to allocating collators etc.

So do you mean just the placement of m_needsTurkishCasingSet = true;?

Will it be ruled out just by using bool? as you mentioned?